app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt

```diff
@@ -867,6 +870,54 @@ class ComposeActivity :
         binding.addPollTextActionTextView.compoundDrawablesRelative[0].colorFilter = PorterDuffColorFilter(textColor, PorterDuff.Mode.SRC_IN)
     }
 
+    // Use to pass state from cropImage caller to result function
+    private var cropImageItemOld: QueuedMedia? = null
+    private var cropImageUriNew: Uri? = null
+    private var cropImageFileNew: File? = null
```

Comment:
I could not find a way to pass "user data" from an ActivityResultContract's caller to the ActivityResult callback. That was a surprising omission, so maybe there is a way in the API I do not know. Because, I had to create these temporary variables to pass the data. If there is a better way to pass the data the cropImage result callback could be simplified.

Reply:
The member variables are lost when the activity is recreated though, e.g. on screen rotation so you should either save them in the ViewModel (preferred) or save & restore them on activity recreation.